pkg/kubectl/cmd/create_pvc.go

```diff
@@ -0,0 +1,210 @@
+	"k8s.io/kubernetes/pkg/api"
+	internalversionstorage "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset/typed/core/internalversion"
+	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/util/i18n"
+)
+
+var (
+	pvcLong = templates.LongDesc(i18n.T(`
+		Create a persistent volume claim using specified subcommand`))
+
+	pvcExample = templates.Examples(i18n.T(`
+	    # Create a persistent volume claim named "myclaim" with "ReadWriteOnce" access mode and "1Gi" capacity.
+		kubectl create persistentvolumeclaim myclaim --access-mode=ReadWriteOnce --capacity=1Gi
+
+		# Create a persistent volume claim named "myclaim" with "ReadWriteOnce" access mode, "1Gi" capacity and a storage class named "highquality".
+		kubectl create persistentvolumeclaim myclaim --access-mode=ReadWriteOnce --capacity=1Gi --storage-class=highquality
+
+		# Create a persistent volume claim named "myclaim" with "ReadWriteOnce" access mode, "1Gi" capacity, requiring a persistent volume named "highqualityvolume".
+		# Note: in this case, the created PersistentVolumeClaim will use an exact PersistentVolume which should be created manually by user. This will bypass dynamic provisioning and common binding procedure.
+		kubectl create persistentvolumeclaim myclaim --access-mode=ReadWriteOnce --capacity=1Gi --volume-name=highqualityvolume
+
+		# Create a persistent volume claim named "myclaim" with multiple access modes and "1Gi" capacity.
+		kubectl create persistentvolumeclaim myclaim --access-mode=RWO,RWX --capacity=1Gi`))
+)
+
+type CreatePVCOptions struct {
+	Name         string
+	AccessModes  []api.PersistentVolumeAccessMode
+	Capacity     *resource.Quantity
+	StorageClass *string
+	VolumeName   string
+
+	DryRun       bool
+	OutputFormat string
+	Namespace    string
+	Client       internalversionstorage.PersistentVolumeClaimInterface
+	Mapper       meta.RESTMapper
+	Out          io.Writer
+	PrintObject  func(obj runtime.Object) error
+}
+
+// NewCmdCreatePersistentVolumeClaim is a command to ease creating PersistentVolumeClaims.
+func NewCmdCreatePersistentVolumeClaim(f cmdutil.Factory, cmdOut io.Writer) *cobra.Command {
+	c := &CreatePVCOptions{
+		Out: cmdOut,
+	}
+	cmd := &cobra.Command{
+		Use:     "persistentvolumeclaim NAME --access-mode=accessmode --capacity=capacity [--storage-class=storageclass] [--volume-name=volumename] [--dry-run]",
+		Aliases: []string{"pvc"},
+		Short:   pvcLong,
+		Long:    pvcLong,
+		Example: pvcExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(c.Complete(f, cmd, args))
+			cmdutil.CheckErr(c.Validate())
+			cmdutil.CheckErr(c.RunCreatePVC())
+		},
+	}
+	cmdutil.AddApplyAnnotationFlags(cmd)
+	cmdutil.AddValidateFlags(cmd)
+	cmdutil.AddPrinterFlags(cmd)
+	cmdutil.AddDryRunFlag(cmd)
+
+	cmd.Flags().StringSlice("access-mode", []string{}, "AccessMode contains the desired access modes the volume should have.")
```

Comment:
perhaps `comma-separated list of required access modes that the desired persistent volume must have`. Or anything that would point out that it is actually list and comma is the separator.

Reply:
this never got resolved

Reply:
Have updated the option description, PTAL.

Reply:
fine for me, thanks